iina/Shared.xcconfig

```diff
@@ -0,0 +1,56 @@
+//
+//  Shared.xcconfig
+//  iina
+//
+//  Created by Saagar Jha on 11/5/22.
+//  Copyright © 2022 lhc. All rights reserved.
+//
+
+#include "Deployment.xcconfig"
+
+ALWAYS_SEARCH_USER_PATHS = NO
+CLANG_ANALYZER_NONNULL = YES
+CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION = YES_AGGRESSIVE
+CLANG_CXX_LANGUAGE_STANDARD = gnu++17
+CLANG_CXX_LIBRARY = libc++
+CLANG_ENABLE_MODULES = YES
+CLANG_ENABLE_OBJC_ARC = YES
+CLANG_ENABLE_OBJC_WEAK = YES
+CLANG_WARN__DUPLICATE_METHOD_MATCH = YES
+CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES
+CLANG_WARN_BOOL_CONVERSION = YES
+CLANG_WARN_COMMA = YES
+CLANG_WARN_CONSTANT_CONVERSION = YES
+CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES
+CLANG_WARN_DIRECT_OBJC_ISA_USAGE = YES_ERROR
+CLANG_WARN_DOCUMENTATION_COMMENTS = YES
```

Comment:
I'm getting a bunch of new warnings such as "Not a Doxygen trailing comment". If I change this to "NO" the new warnings are gone.

Searching the iina.xcodeproj file in develop (not this PR) I see:
```text
low-batt@gag iina.xcodeproj (develop $=)$ grep COMMENT project.pbxproj 
				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
low-batt@gag iina.xcodeproj (develop $=)$ 
```

However searching for "comment" in the Xcode build setting using the UI I see all "no" and no "yes":
![pr-4041](https://user-images.githubusercontent.com/86170219/200203191-0251eb54-3cb2-474c-a504-792a6619094a.jpg)

I'm confused as to where the "no" settings are coming from. But I wasn't getting these warnings before.

Reply:
The new settings don't precisely match the old ones, I stole them from a fresh project and then diffed them to get a set that would be appropriate for IINA (i.e. I picked the customized ones that we wanted, and reverted to Xcode's default for ones that were probably counterproductive). For warnings I tended towards enabling them. In this case, I guess we should either look at getting the comments to match up or temporarily disabling this setting.

Reply:
From what I remember the offending comments are in FFmpeg include files. We'd have to get that project to change their sources, or apply patches each time we update FFmpeg.

I too prefer to enable most warnings. In some projects I've enabled "treat warnings as errors" as it was the only way to get some developers to stop committing sloppy code that introduced warnings.

Reply:
>  We'd have to get that project to change their sources, or apply patches each time we update FFmpeg.

I'll look into this. Let's keep it in for now and if there isn't any progress we'll turn it off until they fix it.

> In some projects I've enabled "treat warnings as errors" as it was the only way to get some developers to stop committing sloppy code that introduced warnings.

I trust you all more than that :)